### ISSUE TITLE
 Questions on how to build the raymarching

### ISSUE THREAD

**Issue by daichang01** [facebookresearch/pytorch3d#1672]
## ❓ Questions on how to build the raymarching
The environment I installed in conda is as follows：
(nerfmesh) PS E:\nerf2mesh-0.2.0\raymarching> conda list
# packages in environment at D:\miniconda\envs\nerfmesh:
#
# Name                    Version                   Build  Channel
asttokens                 2.4.0              pyhd8ed1ab_0    conda-forge
backcall                  0.2.0              pyh9f0ad1d_0    conda-forge
backports                 1.0                pyhd8ed1ab_3    conda-forge
backports.functools_lru_cache 1.6.5              pyhd8ed1ab_0    conda-forge
bzip2                     1.0.8                he774522_0    defaults
ca-certificates           2023.08.22           haa95532_0    defaults
certifi                   2023.7.22                pypi_0    pypi
charset-normalizer        3.3.1                    pypi_0    pypi
colorama                  0.4.6              pyhd8ed1ab_0    conda-forge
comm                      0.1.4              pyhd8ed1ab_0    conda-forge
contourpy                 1.1.1                    pypi_0    pypi
cuda-cccl                 11.6.55                       0    nvidia/label/cuda-11.6.0
cuda-command-line-tools   11.6.0                        0    nvidia/label/cuda-11.6.0
cuda-compiler             11.6.0                        0    nvidia/label/cuda-11.6.0
cuda-cudart               11.6.55                       0    nvidia/label/cuda-11.6.0
cuda-cudart-dev           11.6.55                       0    nvidia/label/cuda-11.6.0
cuda-cuobjdump            11.6.55                       0    nvidia/label/cuda-11.6.0
cuda-cupti                11.6.55                       0    nvidia/label/cuda-11.6.0
cuda-cuxxfilt             11.6.55                       0    nvidia/label/cuda-11.6.0
cuda-libraries            11.6.0                        0    nvidia/label/cuda-11.6.0
cuda-libraries-dev        11.6.0                        0    nvidia/label/cuda-11.6.0
cuda-memcheck             11.6.55                       0    nvidia/label/cuda-11.6.0
cuda-nsight-compute       11.6.0                        0    nvidia/label/cuda-11.6.0
cuda-nvcc                 11.6.55                       0    nvidia/label/cuda-11.6.0
cuda-nvdisasm             11.6.55                       0    nvidia/label/cuda-11.6.0
cuda-nvml-dev             11.6.55                       0    nvidia/label/cuda-11.6.0
cuda-nvprof               11.6.55                       0    nvidia/label/cuda-11.6.0
cuda-nvprune              11.6.55                       0    nvidia/label/cuda-11.6.0
cuda-nvrtc                11.6.55                       0    nvidia/label/cuda-11.6.0
cuda-nvrtc-dev            11.6.55                       0    nvidia/label/cuda-11.6.0
cuda-nvtx                 11.6.55                       0    nvidia/label/cuda-11.6.0
cuda-nvvp                 11.6.58                       0    nvidia/label/cuda-11.6.0
cuda-sanitizer-api        11.6.55                       0    nvidia/label/cuda-11.6.0
cuda-toolkit              11.6.0                        0    nvidia/label/cuda-11.6.0
cuda-tools                11.6.0                        0    nvidia/label/cuda-11.6.0
cuda-visual-tools         11.6.0                        0    nvidia/label/cuda-11.6.0
cycler                    0.12.1                   pypi_0    pypi
dearpygui                 1.10.1                   pypi_0    pypi
debugpy                   1.6.7           py310hd77b12b_0    defaults
decorator                 5.1.1              pyhd8ed1ab_0    conda-forge
exceptiongroup            1.1.3              pyhd8ed1ab_0    conda-forge
executing                 1.2.0              pyhd8ed1ab_0    conda-forge
filelock                  3.12.4                   pypi_0    pypi
fonttools                 4.43.1                   pypi_0    pypi
fsspec                    2023.10.0                pypi_0    pypi
fvcore                    0.1.5.post20221221          pypi_0    pypi
idna                      3.4                      pypi_0    pypi
imageio                   2.31.6                   pypi_0    pypi
imageio-ffmpeg            0.4.9                    pypi_0    pypi
importlib-metadata        6.8.0              pyha770c72_0    conda-forge
importlib_metadata        6.8.0                hd8ed1ab_0    conda-forge
iopath                    0.1.10                   pypi_0    pypi
ipykernel                 6.25.2             pyh60829e3_0    conda-forge
ipython                   8.15.0          py310haa95532_0    defaults
jedi                      0.19.1             pyhd8ed1ab_0    conda-forge
jinja2                    3.1.2                    pypi_0    pypi
joblib                    1.3.2                    pypi_0    pypi
jupyter_client            8.4.0              pyhd8ed1ab_0    conda-forge
jupyter_core              5.3.0           py310haa95532_0    defaults
kiwisolver                1.4.5                    pypi_0    pypi
libcublas                 11.8.1.74                     0    nvidia/label/cuda-11.6.0
libcublas-dev             11.8.1.74                     0    nvidia/label/cuda-11.6.0
libcufft                  10.7.0.55                     0    nvidia/label/cuda-11.6.0
libcufft-dev              10.7.0.55                     0    nvidia/label/cuda-11.6.0
libcurand                 10.2.9.55                     0    nvidia/label/cuda-11.6.0
libcurand-dev             10.2.9.55                     0    nvidia/label/cuda-11.6.0
libcusolver               11.3.2.55                     0    nvidia/label/cuda-11.6.0
libcusolver-dev           11.3.2.55                     0    nvidia/label/cuda-11.6.0
libcusparse               11.7.1.55                     0    nvidia/label/cuda-11.6.0
libcusparse-dev           11.7.1.55                     0    nvidia/label/cuda-11.6.0
libffi                    3.4.4                hd77b12b_0    defaults
libnpp                    11.6.0.55                     0    nvidia/label/cuda-11.6.0
libnpp-dev                11.6.0.55                     0    nvidia/label/cuda-11.6.0
libnvjpeg                 11.6.0.55                     0    nvidia/label/cuda-11.6.0
libnvjpeg-dev             11.6.0.55                     0    nvidia/label/cuda-11.6.0
libsodium                 1.0.18               h8d14728_1    conda-forge
lightning-utilities       0.9.0                    pypi_0    pypi
lpips                     0.1.4                    pypi_0    pypi
markdown-it-py            3.0.0                    pypi_0    pypi
markupsafe                2.1.3                    pypi_0    pypi
matplotlib                3.8.0                    pypi_0    pypi
matplotlib-inline         0.1.6              pyhd8ed1ab_0    conda-forge
mdurl                     0.1.2                    pypi_0    pypi
mpmath                    1.3.0                    pypi_0    pypi
msvc-runtime              14.34.31931              pypi_0    pypi
nest-asyncio              1.5.8              pyhd8ed1ab_0    conda-forge
networkx                  3.2                      pypi_0    pypi
ninja                     1.11.1.1                 pypi_0    pypi
nsight-compute            2022.1.0.12                   0    nvidia/label/cuda-11.6.0
numpy                     1.26.1                   pypi_0    pypi
nvdiffrast                0.3.1                    pypi_0    pypi
opencv-python             4.8.1.78                 pypi_0    pypi
openssl                   3.0.11               h2bbff1b_2    defaults
packaging                 23.2               pyhd8ed1ab_0    conda-forge
pandas                    2.1.1                    pypi_0    pypi
parso                     0.8.3              pyhd8ed1ab_0    conda-forge
pickleshare               0.7.5                   py_1003    conda-forge
pillow                    10.0.1                   pypi_0    pypi
pip                       23.3            py310haa95532_0    defaults
platformdirs              3.11.0             pyhd8ed1ab_0    conda-forge
portalocker               2.8.2                    pypi_0    pypi
prompt-toolkit            3.0.39             pyha770c72_0    conda-forge
prompt_toolkit            3.0.39               hd8ed1ab_0    conda-forge
protobuf                  4.24.4                   pypi_0    pypi
psutil                    5.9.0           py310h2bbff1b_0    defaults
pure_eval                 0.2.2              pyhd8ed1ab_0    conda-forge
pygments                  2.16.1             pyhd8ed1ab_0    conda-forge
pymcubes                  0.1.4                    pypi_0    pypi
pymeshlab                 2022.2.post4             pypi_0    pypi
pyparsing                 3.1.1                    pypi_0    pypi
python                    3.10.13              he1021f5_0    defaults
python-dateutil           2.8.2              pyhd8ed1ab_0    conda-forge
pytorch3d                 0.7.4                    pypi_0    pypi
pytz                      2023.3.post1             pypi_0    pypi
pywin32                   305             py310h2bbff1b_0    defaults
pyyaml                    6.0.1                    pypi_0    pypi
pyzmq                     25.1.0          py310hd77b12b_0    defaults
requests                  2.31.0                   pypi_0    pypi
rich                      13.6.0                   pypi_0    pypi
scikit-learn              1.3.2                    pypi_0    pypi
scipy                     1.11.3                   pypi_0    pypi
setuptools                68.0.0          py310haa95532_0    defaults
six                       1.16.0             pyh6c4a22f_0    conda-forge
sqlite                    3.41.2               h2bbff1b_0    defaults
stack_data                0.6.2              pyhd8ed1ab_0    conda-forge
sympy                     1.12                     pypi_0    pypi
tabulate                  0.9.0                    pypi_0    pypi
tensorboardx              2.6.2.2                  pypi_0    pypi
termcolor                 2.3.0                    pypi_0    pypi
threadpoolctl             3.2.0                    pypi_0    pypi
tinycudann                1.7                      pypi_0    pypi
tk                        8.6.12               h2bbff1b_0    defaults
torch                     1.12.1+cu116             pypi_0    pypi
torch-ema                 0.3                      pypi_0    pypi
torch-scatter             2.1.0+pt112cu116          pypi_0    pypi
torchaudio                0.12.1+cu116             pypi_0    pypi
torchmetrics              1.2.0                    pypi_0    pypi
torchvision               0.13.1+cu116             pypi_0    pypi
tornado                   6.3.3           py310h2bbff1b_0    defaults
tqdm                      4.66.1                   pypi_0    pypi
traitlets                 5.11.2             pyhd8ed1ab_0    conda-forge
trimesh                   4.0.0                    pypi_0    pypi
typing-extensions         4.8.0                hd8ed1ab_0    conda-forge
typing_extensions         4.8.0              pyha770c72_0    conda-forge
tzdata                    2023.3                   pypi_0    pypi
urllib3                   2.0.7                    pypi_0    pypi
vc                        14.2                 h21ff451_1    defaults
vs2015_runtime            14.27.29016          h5e58377_2    defaults
wcwidth                   0.2.8              pyhd8ed1ab_0    conda-forge
wheel                     0.41.2          py310haa95532_0    defaults
xatlas                    0.0.8                    pypi_0    pypi
xz                        5.4.2                h8cc25b3_0    defaults
yacs                      0.1.8                    pypi_0    pypi
zeromq                    4.3.4                h0e60522_1    conda-forge
zipp                      3.17.0             pyhd8ed1ab_0    conda-forge
zlib                      1.2.13               h8cc25b3_0    defaults
Visual studio2019 used

**When I build raymarching I get the following error**：

(nerfmesh) PS E:\nerf2mesh-0.2.0\raymarching> python .\setup.py build_ext --inplace
running build_ext
building '_raymarching_mob' extension
Emitting ninja build file E:\nerf2mesh-0.2.0\raymarching\build\temp.win-amd64-cpython-310\Release\build.ninja...
Compiling objects...
Allowing ninja to set a default number of workers... (overridable by setting the environment variable MAX_JOBS=N)
1.11.1.git.kitware.jobserver-1
"C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30133\bin\HostX86\x64\link.exe" /nologo /INCREMENTAL:NO /LTCG /DLL /MANIFEST:EMBED,ID=2 /MANIFESTUAC:NO /LIBPATH:D:\miniconda\envs\nerfmesh\lib\site-packages\torch\lib /LIBPATH:D:\miniconda\envs\nerfmesh\lib/x64 /LIBPATH:D:\miniconda\envs\nerfmesh\libs /LIBPATH:D:\miniconda\envs\nerfmesh /LIBPATH:D:\miniconda\envs\nerfmesh\PCbuild\amd64 "/LIBPATH:C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30133\lib\x64" "/LIBPATH:C:\Program Files (x86)\Windows Kits\10\lib\10.0.22000.0\ucrt\x64" "/LIBPATH:C:\Program Files (x86)\Windows Kits\10\lib\10.0.22000.0\um\x64" "/LIBPATH:C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30133\lib\x64" c10.lib torch.lib torch_cpu.lib torch_python.lib cudart.lib c10_cuda.lib torch_cuda_cu.lib torch_cuda_cpp.lib /EXPORT:PyInit__raymarching_mob E:\nerf2mesh-0.2.0\raymarching\build\temp.win-amd64-cpython-310\Release\nerf2mesh-0.2.0\raymarching\src\bindings.obj E:\nerf2mesh-0.2.0\raymarching\build\temp.win-amd64-cpython-310\Release\nerf2mesh-0.2.0\raymarching\src\raymarching.obj /OUT:build\lib.win-amd64-cpython-310\_raymarching_mob.cp310-win_amd64.pyd /IMPLIB:E:\nerf2mesh-0.2.0\raymarching\build\temp.win-amd64-cpython-310\Release\nerf2mesh-0.2.0\raymarching\src\_raymarching_mob.cp310-win_amd64.lib
LINK : fatal error LNK1181: 无法打开输入文件“E:\nerf2mesh-0.2.0\raymarching\build\temp.win-amd64-cpython-310\Release\nerf2mesh-0.2.0\raymarching\src\bindings.obj”
error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.29.30133\\bin\\HostX86\\x64\\link.exe' failed with exit code 1181

**Comment by bottler** [comment]
I don't think this is a PyTorch3D thing. But it may be that you need to install more components of visual studio, e.g. C++ build tools.
